rust/xaynet-analytics/src/database/analytics_event/data_model.rs

```diff
@@ -13,55 +16,56 @@ pub enum AnalyticsEventType {
 }
 
 #[derive(Debug, PartialEq, Eq, Clone)]
-pub struct AnalyticsEvent {
+pub struct AnalyticsEvent<'a> {
     pub name: String,
     pub event_type: AnalyticsEventType,
     pub timestamp: DateTime<Utc>,
-    pub screen_route: Option<String>,
+    pub screen_route: Option<&'a ScreenRoute>,
 }
 
-impl AnalyticsEvent {
-    pub fn new<N: Into<String>, R: Into<Option<String>>>(
+impl<'a> AnalyticsEvent<'a> {
+    pub fn new<N: Into<String>>(
         name: N,
         event_type: AnalyticsEventType,
         timestamp: DateTime<Utc>,
-        screen_route: R,
+        screen_route: Option<&'a ScreenRoute>,
     ) -> AnalyticsEvent {
         AnalyticsEvent {
             name: name.into(),
             event_type,
             timestamp,
-            screen_route: screen_route.into(),
+            screen_route,
         }
     }
 
-    fn add_screen_route(&self, object_builder: &mut ObjectBuilder) {
+    fn write_screen_route(&self, object_builder: &mut ObjectBuilder) {
         match &self.screen_route {
-            Some(screen) => object_builder.write_string(Some(&screen)),
+            Some(screen) => match &screen.object_id {
+                Some(object_id) => object_builder.write_string(Some(&object_id)),
```

Comment:
What is the difference between calling `write_string(None)` and `write_null()` ?

Reply:
well eventually nothing, cause `write_string(None)` is wrapped in `write_null()` for this `DataType`, but for other `DataType`s is not as straightforward, so i thought we'd use what the public API offers

Reply:
Because if it is the same we just can write `write_string(screen_route.map(|screen| &screen.object_id)` and let isar do the job.

Reply:
yeh i know. it's nice like this, but it might be an issue for other DataTypes, so i thought i'd be more explicit and avoid other devs get into weird crashes for not using `write_null()` in the future

Reply:
ok after we talked (for the record), i'll change it. thanks

Reply:
ok done but i used the name, since the object_id is constructed from that. we'll see with the `get_all` implementation if it works, i guess